File: Devashree Ambegaokar/Array/2 Two sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> my_map;

        for (int i = 0; i < nums.size(); i++) {
            int complement = target - nums[i];

            if (my_map.find(complement) != my_map.end()) //check karne ke liye if the number is in the array ya nahi
            {
                return {i, my_map[complement]}; 
            }
            my_map[nums[i]] = i;
        }
        return{};
    }
};

File: Riya khandelwal/Array/2.Two_sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int i=0;
        int j=nums.length-1;
        int[] ind = new int[nums.length];
        for(int i=0;i<nums.length;i++)
        {
            in
        }
        Arrays.sort(nums);
        int[] arr = new int[2];
        while(i<j)
        {
            if(nums[i]+nums[j]>target)
            {
                j--;
            }
            else if (nums[i]+nums[j]<target)
            {
                i++;
            }
            else
            {
                arr[0] = i;
                arr[1] = j;
            }

        }
        return arr;
    }
}
